tests/end_to_end/test_helper.cpp

```diff
@@ -534,9 +534,24 @@ _test_helper_end_to_end::_test_helper_end_to_end()
     api_initialized = true;
 }
 
+static void
+_rundown_osfhandles()
+{
+    std::vector<int> fds_to_close;
+    for (auto [fd, handle] : _fd_to_handle_map) {
+        fds_to_close.push_back(fd);
```

Comment:
Can we directly call `Glue_close(fd)` here?

Reply:
No, because Glue_close will invalidate the iterator we are using to walk the map, hence gathering all the fds first.

Reply:
I had the same question.  Please add a // comment explaining this.